test/WebJobs.Script.Tests/Workers/Rpc/RpcFunctionInvocationDispatcherTests.cs

```diff
@@ -549,6 +549,111 @@ public async Task FunctionDispatcher_Restart_ErroredChannels_OnWorkerRestart_Not
+            var invocationId = Guid.NewGuid().ToString();
+            var mockChannel = new Mock<IRpcWorkerChannel>(MockBehavior.Strict);
+            var mockJobHostChannelManager = new Mock<IJobHostRpcWorkerChannelManager>(MockBehavior.Strict);
+            var mockWebHostChannelManager = new Mock<IWebHostRpcWorkerChannelManager>(MockBehavior.Strict);
+
+            // Setup the channel managers to return our mock channel
+            mockWebHostChannelManager.Setup(mockWebHostChannelManager => mockWebHostChannelManager.GetChannels(It.IsAny<string>()))
+                                     .Returns(new Dictionary<string, TaskCompletionSource<IRpcWorkerChannel>>());
+            mockJobHostChannelManager.Setup(m => m.GetChannels(It.IsAny<string>()))
+                  .Returns(new List<IRpcWorkerChannel> { mockChannel.Object });
+            mockJobHostChannelManager.Setup(m => m.GetChannels())
+                              .Returns(new List<IRpcWorkerChannel> { mockChannel.Object });
+
+            // Setup the mock channel to indicate it's executing the specified invocation
+            mockChannel.Setup(c => c.Id).Returns("testChannelId");
+            mockChannel.Setup(c => c.IsExecutingInvocation(invocationId)).Returns(true);
+            mockChannel.Setup(c => c.IsChannelReadyForInvocations()).Returns(true);
+
+            // Set up ShutdownChannelIfExistsAsync to be called with the right exception type
+            mockWebHostChannelManager.Setup(m => m.ShutdownChannelIfExistsAsync(
+                It.IsAny<string>(),
+                It.Is<string>(id => id == "testChannelId"),
+                It.Is<Exception>(ex => ex is TimeoutException && ex.Message == $"Executing invocation `{invocationId}` timed out")))
+                .ReturnsAsync(true);
+
+            // Setup for initialization of a new channel
+            var workerConfig = new RpcWorkerConfig
+            {
+                Description = new RpcWorkerDescription { Language = "test" },
+                CountOptions = new WorkerProcessCountOptions { ProcessCount = 1 }
+            };
+
+            // Create the dispatcher with our mocks
+            var dispatcher = GetTestFunctionDispatcher(mockwebHostLanguageWorkerChannelManager: mockWebHostChannelManager, mockJobHostLanguageWorkerChannelManager: mockJobHostChannelManager);
+
+            // Act
+            var result = await dispatcher.RestartWorkerWithInvocationIdAsync(invocationId);
+
+            // Assert
+            Assert.True(result);
+            mockWebHostChannelManager.Verify(m => m.ShutdownChannelIfExistsAsync(
+                It.IsAny<string>(),
+                It.Is<string>(id => id == "testChannelId"),
+                It.Is<Exception>(ex => ex is TimeoutException && ex.Message == $"Executing invocation `{invocationId}` timed out")),
+                Times.Once);
+        }
+
+        [Fact]
+        public async Task FunctionDispatcher_RestartOfTimedOutChannels_JobHostFailsCurrentExecutions()
```

Comment:
The WebHost will try to shut down the channel first and fail executions, but in the case that returns false we want to make sure the JobHost attempts to. See [here](https://github.com/Azure/azure-functions-host/blob/49e1420b87b9d4a867089c968cddeb5eca288cc0/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs#L540C15-L544C18).